### DISCUSSION TITLE
Cannot set priors for hierarchical ddm

### DISCUSSION THREAD

**Discussion by TeaforTwo1011** [lnccbrown/HSSM#365] [Q&A] [answered]
Dear HSSM team, 

I am trying to fit a hierarchical ddm to my data, where I start with the model:

```
model_ddm = hssm.HSSM(
    data = my_data, 
    model="ddm",
    loglik_kind="approx_differentiable",
    include=[
        {   "name":"a",
            "formula": "a ~ 0 + (1|Sub)",
            "prior":{"name":"uniform", "lower":0, "upper":4}
            },
        {
            "name":"v",
            "formula":"v ~ 0 + (0 + S|Sub)",
            "prior":{"name":"normal", "mu": 0, "sigma": 0.01}
            },
        {   "name":"z",
            "formula": "z ~ 0 + (1|Sub)",
            "prior":{"name":"uniform", "lower":0.01, "upper":0.99}
            },
        {   "name":"t",
            "formula": "t ~ 0 + (1|Sub)",
            "prior":{"name":"uniform", "lower":0.01, "upper":3}
        }]
                  )
```

Here I specified the priors, however, when I print the model with
`model_ddm`,
what I get is:

```
Hierarchical Sequential Sampling Model
Model: ddm

Response variable: rt,response
Likelihood: approx_differentiable
Observations: 19110

Parameters:

v:
    Formula: v ~ 0 + (0 + S|Sub)
    Priors:
        v_S|Sub ~ Normal(mu: 0.0, sigma: HalfNormal(sigma: [5.0001 5.0001]))
    Link: identity
    Explicit bounds: (-3.0, 3.0)
a:
    Formula: a ~ 0 + (1|Sub)
    Priors:
        a_1|Sub ~ Normal(mu: 0.0, sigma: HalfNormal(sigma: 1.0))
    Link: identity
    Explicit bounds: (0.3, 2.5)
z:
    Formula: z ~ 0 + (1|Sub)
    Priors:
        z_1|Sub ~ Normal(mu: 0.0, sigma: HalfNormal(sigma: 1.0))
    Link: identity
    Explicit bounds: (0.0, 1.0)
t:
    Formula: t ~ 0 + (1|Sub)
    Priors:
        t_1|Sub ~ Normal(mu: 0.0, sigma: HalfNormal(sigma: 1.0))
    Link: identity
    Explicit bounds: (0.0, 2.0)

Lapse probability: 0.05
Lapse distribution: Uniform(lower: 0.0, upper: 10.0)
```

This does not match what I have supplied in the `include=` argument.

Could you let me know what else should be done here?

Thanks a lot for your help!

Best,
Yufei

**Comment by digicosmos86** [comment, marked as answer]
Hi @TeaforTwo1011,

Could you provide a bit more details about what model are you trying to build?

From the code that you have shared, you have specified regressions for each parameter, therefore, you need to specify priors for each regression term. Let's say that you are specifying a regression on v: `v ~ 1 + (1|subj_id)`, then in the corresponding dictionary in `include`, you need to specify priors like this:

```
model = hssm.HSSM(
    model="ddm",
    include=[
        {
            "name": "v",
            "formula": "v ~ 1 + (1|subj_id)",
            "priors": {
                  "Intercept": {"name": "Uniform", "lower": -2.0, "upper": 2.0},
                  "1|subj_id": {"name": "Normal", "mu": 0, "sigma": {"name": "HalfNormal", "sigma": 2.0}} # You can specify hyperpriors
            }
     ]
)
```

Because you did not specify priors for regression terms, HSSM used default priors for all the regression terms in your model. That's why you are seeing priors that you did not specify. For more details on how to specify priors for hierarchical models, please see [our tutorial here](https://lnccbrown.github.io/HSSM/getting_started/hierarchical_modeling/).

Hope this is helpful! Please let me know if you have any other questions.

Thanks!
Paul
